docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc

```diff
@@ -1464,6 +1464,37 @@ public class DiscoveryEndpointResponseFilter implements OidcResponseFilter {
 <3> Use `OidcRequestContextProperties` request properties to get the tenant id.
 <4> Get the response data as String.
 
+== Programmatic OIDC start-up
+
+OIDC tenants can be created programmatically like in the example below:
+
+[source,java]
+----
+package io.quarkus.it.oidc;
+
+import io.quarkus.oidc.Oidc;
+import io.quarkus.oidc.OidcTenantConfig;
+import jakarta.enterprise.event.Observes;
+
+public class OidcStartup {
+
+    void observe(@Observes Oidc oidc) {
+        oidc.create(OidcTenantConfig.authServerUrl("http://localhost:8180/realms/quarkus").build());
```

Comment:
Proposing a shortcut, `Oidc.create("http://localhost:8180/realms/quarkus")`.

Reply:
This has 2 issues:

- we already have `create` method that is not static, so we can't have one `create` method static and one non-static, both with same signature:
  - if you proposed different name than of course it is trivial to do, we just do inside that static `create` method something like `Arc.container().select(Oidc.class).createWhatver(...)`
- timing is key, using `Oidc.create` must be primarily done before OIDC has started, which is why I have chosen the observer method; I can elaborate why is it so important if it is not clear, just let me know

Reply:
ah, I get it. you don't mind static, you suggest that `oidc.create(`auth-server-url`)` simply builds the tenant right? I misunderstood you

Reply:
Hi @michalvavrik, sure, that was just the first review try.

Yes, sorry, in the `@Observes` case it is `oidc.create(url)`;